src/System.Runtime.Extensions/src/System/Environment.SpecialFolder.cs

```diff
@@ -0,0 +1,115 @@
+            Cookies = SpecialFolderValues.CSIDL_COOKIES,
+            Desktop = SpecialFolderValues.CSIDL_DESKTOP,
+            Favorites = SpecialFolderValues.CSIDL_FAVORITES,
+            History = SpecialFolderValues.CSIDL_HISTORY,
+            InternetCache = SpecialFolderValues.CSIDL_INTERNET_CACHE,
+            Programs = SpecialFolderValues.CSIDL_PROGRAMS,
+            MyComputer = SpecialFolderValues.CSIDL_DRIVES,
+            MyMusic = SpecialFolderValues.CSIDL_MYMUSIC,
+            MyPictures = SpecialFolderValues.CSIDL_MYPICTURES,
+            MyVideos = SpecialFolderValues.CSIDL_MYVIDEO,
+            Recent = SpecialFolderValues.CSIDL_RECENT,
+            SendTo = SpecialFolderValues.CSIDL_SENDTO,
+            StartMenu = SpecialFolderValues.CSIDL_STARTMENU,
+            Startup = SpecialFolderValues.CSIDL_STARTUP,
+            System = SpecialFolderValues.CSIDL_SYSTEM,
+            Templates = SpecialFolderValues.CSIDL_TEMPLATES,
+            DesktopDirectory = SpecialFolderValues.CSIDL_DESKTOPDIRECTORY,
+            Personal = SpecialFolderValues.CSIDL_PERSONAL,
+            MyDocuments = SpecialFolderValues.CSIDL_PERSONAL,
+            ProgramFiles = SpecialFolderValues.CSIDL_PROGRAM_FILES,
+            CommonProgramFiles = SpecialFolderValues.CSIDL_PROGRAM_FILES_COMMON,
+            AdminTools = SpecialFolderValues.CSIDL_ADMINTOOLS,
+            CDBurning = SpecialFolderValues.CSIDL_CDBURN_AREA,
+            CommonAdminTools = SpecialFolderValues.CSIDL_COMMON_ADMINTOOLS,
+            CommonDocuments = SpecialFolderValues.CSIDL_COMMON_DOCUMENTS,
+            CommonMusic = SpecialFolderValues.CSIDL_COMMON_MUSIC,
+            CommonOemLinks = SpecialFolderValues.CSIDL_COMMON_OEM_LINKS,
+            CommonPictures = SpecialFolderValues.CSIDL_COMMON_PICTURES,
+            CommonStartMenu = SpecialFolderValues.CSIDL_COMMON_STARTMENU,
+            CommonPrograms = SpecialFolderValues.CSIDL_COMMON_PROGRAMS,
+            CommonStartup = SpecialFolderValues.CSIDL_COMMON_STARTUP,
+            CommonDesktopDirectory = SpecialFolderValues.CSIDL_COMMON_DESKTOPDIRECTORY,
+            CommonTemplates = SpecialFolderValues.CSIDL_COMMON_TEMPLATES,
+            CommonVideos = SpecialFolderValues.CSIDL_COMMON_VIDEO,
+            Fonts = SpecialFolderValues.CSIDL_FONTS,
+            NetworkShortcuts = SpecialFolderValues.CSIDL_NETHOOD,
+            PrinterShortcuts = SpecialFolderValues.CSIDL_PRINTHOOD,
+            UserProfile = SpecialFolderValues.CSIDL_PROFILE,
+            CommonProgramFilesX86 = SpecialFolderValues.CSIDL_PROGRAM_FILES_COMMONX86,
+            ProgramFilesX86 = SpecialFolderValues.CSIDL_PROGRAM_FILESX86,
+            Resources = SpecialFolderValues.CSIDL_RESOURCES,
+            LocalizedResources = SpecialFolderValues.CSIDL_RESOURCES_LOCALIZED,
+            SystemX86 = SpecialFolderValues.CSIDL_SYSTEMX86,
+            Windows = SpecialFolderValues.CSIDL_WINDOWS,
```

Comment:
I love that some of this stuff is coming back and it will be easier to port, but I wonder if any of these things make sense... Especially on osx and linux.